preload.js

```diff
@@ -50,6 +51,19 @@ window.isBeforeVersion = (toCheck, baseVersion) => {
 
 // temporary clearnet fix
 process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+window.getSelfSignedCert = () => {
+  let pems = window.storage.get('self-signed-certificate', null);
+  if (!pems) {
+    const pubKey = window.storage.get('number_id');
+    const attrs = [{ name: 'commonName', value: pubKey }];
+    pems = selfsigned.generate(attrs, { days: 365 * 10 });
```

Comment:
How long will this take to generate on average?

Reply:
Also you are awaiting this function above, does selfsigned.generate return a promise?

Reply:
`selfsigned.generate` is a sync function, maybe you got confused with `await window.getSelfSignedCert();`?

A quick test in node gives 150-200 ms per call.

Reply:
Sorry that's what I meant, you are awaiting `window.getSelfSignedCert()` but it isn't async so I thought maybe the generate function was returning a promise

You probably don't want to await `window.getSelfSignedCert()` then?

Reply:
Oh I see, you're right. Removed unnecessary awaits.